### PR TITLE
chore: build script updates

### PR DESCRIPTION
**Pull request by broofa**
* ~~Add node@22 to test matrix~~
* Fix `runmd` build
* Change test:pack build dir to /tmp so it doesn't pollute source dir, and is not in-scope of project's package.json

> [!NOTE]
> Reverted the addition of `node@22` to test matrix because the `npm ci` command was failing with the mysterious ["Document exit handler not called" error](https://github.com/npm/statusboard/issues/395).  Given that ci.yml works for node@16,18 and 20, and only fails this way for node@22, and my limited time/interest in tracking this down, I'm going to chalk this up to a node regression and move on.

### CODE REVIEW

File: scripts/testpack.sh [hunk 1]
```diff
@@ -4,20 +4,25 @@
 ROOT="$(pwd)/$(dirname "$0")/.."
 cd "$ROOT" || exit 1
 
-npm pack
+TEST_DIR=/tmp/test-pack
```
Comment (ctavan) [inline scripts/testpack.sh:7]: Prefer mktemp + trap: https://unix.stackexchange.com/a/509189

**Review by ctavan — CHANGES_REQUESTED** [review]

File: scripts/testpack.sh (hunk 1 above)
Comment (broofa) [reply scripts/testpack.sh]: Done.

File: scripts/testpack.sh [hunk 2]
```diff
@@ -4,7 +4,8 @@
 ROOT="$(pwd)/$(dirname "$0")/.."
 cd "$ROOT" || exit 1
 
-TEST_DIR=/tmp/test-pack
+TEST_DIR=$(mktemp -d)
```
Comment (ctavan) [inline scripts/testpack.sh:7]: Set up the trap right after this line so it takes effect if the script exits somewhere in the middle

**Comment by broofa** [comment]
@ctavan Okay to merge?

File: scripts/testpack.sh [hunk 3]
```diff
@@ -4,20 +4,26 @@
...
-npm pack
+TEST_DIR=$(mktemp -d)
 
-mkdir -p ../test-pack
+mkdir -p ${TEST_DIR}
 
-cp examples/node-commonjs/example.js ../test-pack/commonjs.js
-cp examples/node-esmodules/example.mjs ../test-pack/esmodules.mjs
-cp examples/node-esmodules/package.mjs ../test-pack/esmodules-package.mjs
+trap 'popd && rm -rf $TEST_DIR' EXIT
```
Comment (ctavan) [inline scripts/testpack.sh:11]: I’d remove the popd from here… npm pack could fail and then you’d popd without having pushd’ed…

What we mostly care about with the trap is to no leave temp files.

If you want to overengineer, add a separate trap with just popd right after the pushd.

**Review by ctavan — APPROVED** [review]

**Pull request merged**